crates/aptos/src/node/mod.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::types::{CliCommand, CliResult, CliTypedResult, TransactionOptions};
+use aptos_crypto::{ed25519::Ed25519PrivateKey, ValidCryptoMaterialStringExt};
+use aptos_rest_client::Transaction;
+use aptos_types::{account_address::AccountAddress, network_address::NetworkAddress};
+use async_trait::async_trait;
+use clap::Parser;
+
+/// Tool for manipulating nodes
+///
+#[derive(Parser)]
+pub enum NodeTool {
+    AddStake(AddStake),
+    UnlockStake(UnlockStake),
+    WithdrawStake(WithdrawStake),
+    IncreaseLockup(IncreaseLockup),
+    RegisterValidatorCandidate(RegisterValidatorCandidate),
+    RotateConsensusKey(RotateConsensusKey),
+    SetOperator(SetOperator),
+}
+
+impl NodeTool {
+    pub async fn execute(self) -> CliResult {
+        use NodeTool::*;
+        match self {
+            AddStake(tool) => tool.execute_serialized().await,
+            UnlockStake(tool) => tool.execute_serialized().await,
+            WithdrawStake(tool) => tool.execute_serialized().await,
+            IncreaseLockup(tool) => tool.execute_serialized().await,
+            RegisterValidatorCandidate(tool) => tool.execute_serialized().await,
+            RotateConsensusKey(tool) => tool.execute_serialized().await,
+            SetOperator(tool) => tool.execute_serialized().await,
+        }
+    }
+}
+
+/// Stake coins for an account to the stake pool
+#[derive(Parser)]
+pub struct AddStake {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+    /// Amount of coins to add to stake
+    #[clap(long)]
+    pub amount: u64,
```

Comment:
May be something we can do in a future iteration of the CLI - but we should provide some simple balance validation. I.e. that they have enough balance (including gas). This can potentially save the users from a failure that would cost them some gas. I can see this happen for example if they have 10K coins and try to stake all 10K. This leaves them with no balance left for gas, so the that transaction would definitely fail.

Reply:
We should do this on all txns, so we can address it together then